var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -122,31 +125,49 @@ def write_makefile_inc(self):
-        fpic = '-fPIC' if '+shared' in self.spec else ''
+        fpic = self.compiler.pic_flag if '+shared' in self.spec else ''
         # TODO: test this part, it needs a full blas, scalapack and
         # partitionning environment with 64bit integers
+
         if '+int64' in self.spec:
-            makefile_conf.extend(
-                # the fortran compilation flags most probably are
-                # working only for intel and gnu compilers this is
-                # perhaps something the compiler should provide
-                ['OPTF    = %s -O  -DALLOW_NON_INIT %s' % (fpic, '-fdefault-integer-8' if self.compiler.name == "gcc" else '-i8'),  # noqa
-                 'OPTL    = %s -O ' % fpic,
-                 'OPTC    = %s -O -DINTSIZE64' % fpic])
+            if self.compiler.name == "xl" or self.compiler.name == "xl_r":
+                makefile_conf.extend(
+                    ['OPTF    = -O3',
+                     'OPTL    = %s -O3' % fpic,
+                     'OPTC    = %s -O3-DINTSIZE64' % fpic])
+            else:
+                makefile_conf.extend(
+                    # the fortran compilation flags most probably are
+                    # working only for intel and gnu compilers this is
+                    # perhaps something the compiler should provide
+                    ['OPTF    = %s -O  -DALLOW_NON_INIT %s' % (fpic, '-fdefault-integer-8' if self.compiler.name == "gcc" else '-i8'),  # noqa
+                     'OPTL    = %s -O ' % fpic,
+                     'OPTC    = %s -O -DINTSIZE64' % fpic])
         else:
-            makefile_conf.extend(
-                ['OPTF    = %s -O  -DALLOW_NON_INIT' % fpic,
-                 'OPTL    = %s -O ' % fpic,
-                 'OPTC    = %s -O ' % fpic])
+            if self.compiler.name == "xl" or self.compiler.name == "xl_r":
+                makefile_conf.extend(
+                    ['OPTF    = -O3',
+                     'OPTL    = %s -O3' % fpic,
+                     'OPTC    = %s -O3' % fpic])
+            else:
+                makefile_conf.extend(
+                    ['OPTF    = %s -O  -DALLOW_NON_INIT' % fpic,
+                     'OPTL    = %s -O ' % fpic,
+                     'OPTC    = %s -O ' % fpic])
 
         if '+mpi' in self.spec:
             scalapack = self.spec['scalapack'].scalapack_libs
             makefile_conf.extend(
-                ["CC = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpicc'),
-                 "FC = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpif90'),
-                 "FL = %s" % join_path(self.spec['mpi'].prefix.bin, 'mpif90'),
+                ['CC = {0}'.format(self.spec['mpi'].mpicc),
+                 'FC = {0}'.format(self.spec['mpi'].mpifc),
                  "SCALAP = %s" % scalapack.ld_flags,
                  "MUMPS_TYPE = par"])
+            if (self.spec.satisfies('%xl_r' or '%xl')) and self.spec.satisfies('^spectrum-mpi'): # noqa
```

Comment:
Nah, if it works, it works. We should probably document and add tests for this feature if they don't already exist, but that is beyond the scope of this PR.